# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/EncoderTemplate.cpp
# Commit: da42eb3f9664
# Full Hash: da42eb3f9664fc2d5dc77db2dc73e45b025c9a31
# Author: Paul Adenot <paul@paul.cx>
# Date: 2023-12-23 09:52:03
# Regressor Bug: 1749047
# File Overlap Count: 1
# Description:
#   Bug 1749047 - Convert from MediaRawData to EncodedVideoChunk one by one. r=chunmin
#   
#   This simplifies the rest of the code, in remaining patches.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195713
# ==============================================================================

diff -r e59a8dc5add5 -r da42eb3f9664 dom/media/webcodecs/EncoderTemplate.cpp
--- a/dom/media/webcodecs/EncoderTemplate.cpp	Fri Dec 22 21:39:09 2023 +0000
+++ b/dom/media/webcodecs/EncoderTemplate.cpp	Fri Dec 22 21:39:10 2023 +0000
@@ -299,23 +299,23 @@
 }
 
 template <typename EncoderType>
-void EncoderTemplate<EncoderType>::OutputEncodedData(nsTArray<RefPtr<MediaRawData>>&& aData) {
+void EncoderTemplate<EncoderType>::OutputEncodedData(
+    nsTArray<RefPtr<MediaRawData>>&& aData) {
   AssertIsOnOwningThread();
   MOZ_ASSERT(mState == CodecState::Configured);
   MOZ_ASSERT(mActiveConfig);
 
-  nsTArray<RefPtr<typename EncoderType::OutputType>> frames =
-      EncodedDataToOutputType(GetParentObject(), std::move(aData));
   RefPtr<typename EncoderType::OutputCallbackType> cb(mOutputCallback);
-  for (RefPtr<typename EncoderType::OutputType>& frame : frames) {
-    RefPtr<typename EncoderType::OutputType> f = frame;
+  for (auto& data : aData) {
+    RefPtr<typename EncoderType::OutputType> encodedData =
+        EncodedDataToOutputType(GetParentObject(), data);
     typename EncoderType::MetadataType metadata;
     if (mOutputNewDecoderConfig) {
       metadata.mDecoderConfig.Construct(EncoderConfigToDecoderConfig(
-          EncoderConfigToDecoderConfig(GetParentObject(), *mActiveConfig));
+          GetParentObject(), data, *mActiveConfig));
       mOutputNewDecoderConfig = false;
     }
-    cb->Call((typename EncoderType::OutputType&)(*f), metadata);
+    cb->Call((typename EncoderType::OutputType&)(*encodedData), metadata);
   }
 }
 